stdlib/public/core/StringObject.swift

```diff
@@ -449,7 +449,7 @@ extension _StringObject {
   internal static var nativeBias: UInt {
 #if _pointerBitWidth(_64)
     return 32
-#elseif _pointerBitWidth(_32)
+#elseif _pointerBitWidth(_32) || _pointerBitWidth(_16)
     return 20
```

Comment:
This is certainly wrong for 16-bit. Either add a comment with a TODO, or a branch that fatalError()s.

Reply:
I'll put in a TODO. I would like to put in a fatalError as that is more my programming style, but I'm concerned about breaking the unit test suite?

Reply:
By the way, this code in `StringObject` looks possibly buggy for Android AArch64?

```
  @inlinable @inline(__always)
  init(count: Int, variant: Variant, discriminator: UInt64, flags: UInt16) {
#if os(Android) && arch(arm64)
    _internalInvariant(discriminator & 0x00FF_0000_0000_0000 == discriminator,
      "only the second byte can carry the discriminator and small count on Android AArch64")
#else
    _internalInvariant(discriminator & 0xFF00_0000_0000_0000 == discriminator,
      "only the top byte can carry the discriminator and small count")
#endif

    self._count = count
    self._variant = variant
    self._discriminator = UInt8(truncatingIfNeeded: discriminator &>> 56)
    self._flags = flags
    self._invariantCheck()
  }
```

It looks like it should be...
```
#if os(Android) && arch(arm64)
    self._discriminator = UInt8(truncatingIfNeeded: discriminator &>> 48)
#else
    self._discriminator = UInt8(truncatingIfNeeded: discriminator &>> 56)
#endif
```
...shouldn't it?